packages/terminal-extension/src/index.ts

```diff
@@ -274,39 +276,80 @@ export function addCommands(
 
   commands.addCommand(CommandIDs.increaseFont, {
     label: 'Increase Terminal Font Size',
-    execute: () => {
-      let { fontSize } = Terminal.defaultOptions;
+    execute: async () => {
+      let { fontSize } = ITerminal.defaultOptions;
       if (fontSize < 72) {
-        return settingRegistry
-          .set(plugin.id, 'fontSize', fontSize + 1)
-          .catch(showErrorMessage);
+        try {
+          await settingRegistry.set(plugin.id, 'fontSize', fontSize + 1);
+        } catch (err) {
+          showErrorMessage(err);
+        }
       }
     }
   });
 
   commands.addCommand(CommandIDs.decreaseFont, {
     label: 'Decrease Terminal Font Size',
-    execute: () => {
-      let { fontSize } = Terminal.defaultOptions;
+    execute: async () => {
+      let { fontSize } = ITerminal.defaultOptions;
       if (fontSize > 9) {
-        return settingRegistry
-          .set(plugin.id, 'fontSize', fontSize - 1)
-          .catch(showErrorMessage);
+        try {
+          await settingRegistry.set(plugin.id, 'fontSize', fontSize - 1);
+        } catch (err) {
+          showErrorMessage(err);
+        }
       }
     }
   });
 
   commands.addCommand(CommandIDs.toggleTheme, {
     label: 'Use Dark Terminal Theme',
     caption: 'Whether to use the dark terminal theme',
-    isToggled: () => Terminal.defaultOptions.theme === 'dark',
-    execute: () => {
-      let { theme } = Terminal.defaultOptions;
+    isToggled: () => ITerminal.defaultOptions.theme === 'dark',
+    execute: async () => {
+      let { theme } = ITerminal.defaultOptions;
       theme = theme === 'dark' ? 'light' : 'dark';
-      return settingRegistry
-        .set(plugin.id, 'theme', theme)
-        .then(() => commands.notifyCommandChanged(CommandIDs.toggleTheme))
-        .catch(showErrorMessage);
+      try {
+        await settingRegistry.set(plugin.id, 'theme', theme);
+        commands.notifyCommandChanged(CommandIDs.toggleTheme);
+      } catch (err) {
+        showErrorMessage(err);
+      }
     }
   });
 }
+
+/**
+ * A namespace for private data.
+ */
+namespace Private {
+  /**
+   * A Promise for the initial load of the terminal widget.
+   */
+  export let widgetReady: Promise<typeof WidgetModuleType>;
+
+  /**
+   * Lazy-load the widget (and xterm library and addons)
+   */
+  export function ensureWidget(): Promise<typeof WidgetModuleType> {
+    if (widgetReady) {
+      return widgetReady;
+    }
+
+    widgetReady = new Promise((resolve, reject) => {
```

Comment:
again, i really wish this could be made more general, but we're in ugly webpack-specific territory here... `await import` will be way better.